packages/react/src/runtimes/remote-thread-list/RemoteThreadListThreadListRuntimeCore.tsx

```
@@ -0,0 +1,386 @@
+      this._state.baseValue.newThreadId !== undefined &&
+      this._state.value.newThreadId === undefined
+    ) {
+      await this._state.waitForUpdate();
+    }
+
+    const state = this._state.value;
+    let threadId: string | undefined = this._state.value.newThreadId;
+    if (threadId === undefined) {
+      do {
+        threadId = "__LOCALID_" + generateId();
+      } while (!!state.threadData[threadId]);
```

Biome (1.9.4)

[error] Avoid redundant double-negation.

It is not necessary to use double-negation when a value will already be coerced to a boolean.
Unsafe fix: Remove redundant double-negation


(lint/complexity/noExtraBooleanCast)